lib/MusicBrainz/Server/WebService/Serializer/JSON/LD/Artist.pm

```diff
@@ -41,11 +41,20 @@ around serialize => sub {
         if (@members) {
             my %seen_members;
             for my $member (@members) {
-                $seen_members{$member->target->gid} = member_relationship($member, $inc, $stash, $seen_members{$member->target->gid});
+                $seen_members{$member->target->gid} = member_relationship($member, 'member', $inc, $stash, $seen_members{$member->target->gid});
             }
             $ret->{member} = [ map { $seen_members{$_} } sort keys %seen_members ];
         }
 
+        my @member_ofs = grep { $_->direction == 1 } @{ $entity->relationships_by_link_type_names('member of band') };
```

Comment:
Since this runs on every artist page, I'm going to be inquisitive here. :) Can't we combine these blocks instead of running `relationships_by_link_type_names` twice and looping over everything twice? e.g. remove the `grep { $_->direction == $x }` from the block above and check the direction inside the loop?

Reply:
If that doesn't make sense, at least save `@{ $entity->relationships_by_link_type_names('member of band') }` to a variable so both block can use it.

Reply:
Fair enough, made the change :)